tracer/src/Datadog.Trace/Logging/DirectSubmission/Formatting/LogFormatter.cs

```diff
@@ -333,10 +333,20 @@ internal void FormatCIVisibilityLog(StringBuilder sb, string source, string? log
                 }
 
                 writer.WritePropertyName("dd.trace_id", escape: false);
-                writer.WriteValue($"{span.TraceId}");
+
+                if (span is Span { TraceId128.Upper: > 0 } s)
+                {
+                    // 128-bit trace ids are encoded as hex
+                    writer.WriteValue(s.Context.RawTraceId);
+                }
+                else
+                {
+                    // 64-bit trace ids and span ids are encoded as decimal
+                    writer.WriteValue(span.TraceId);
+                }
 
                 writer.WritePropertyName("dd.span_id", escape: false);
-                writer.WriteValue($"{span.SpanId}");
+                writer.WriteValue(span.SpanId);
```

Comment:
We might need this to be a string for logs injection to occur properly, but we'll need to track down some docs or some contacts to validate this theory

Reply:
Oh! In that case, shouldn't we use `span.SpanId.ToString(InvariantCulture)`?
Aside from the culture issue, when targeting older runtimes, this becomes `string.Format("{0}", span.SpanId)`, and the format string `"{0}"` must be parsed at run time.

Reply:
Ohh yeah it sounds like `span.SpanId.ToString(InvariantCulture)` would be an improvement 👍🏼

Reply:
done here and in the other logger